test/test.py

```diff
@@ -32,6 +33,47 @@ def test_pytorch():
     for x,y in zip(test_tinygrad(), test_pytorch()):
       np.testing.assert_allclose(x, y, atol=1e-5)
 
+  def test_gradcheck(self):
+    class TinyModel:
+      def __init__(self, weights_init):
+        self.l1 = Tensor(weights_init)
+      def forward(self, x):
+        return x.dot(self.l1).relu().logsoftmax()
+
+    class TorchModel(torch.nn.Module):
+      def __init__(self, weights_init):
+        super(TorchModel, self).__init__()
+        self.l1 = torch.nn.Linear(*weights_init.shape, bias = False)
+        self.l1.weight = torch.nn.Parameter(torch.tensor(weights_init.T, requires_grad = True))
+      def forward(self, x):
+        return torch.nn.functional.log_softmax(self.l1(x).relu(), dim=1)
+
+    layer_weights = np.random.RandomState(1337).random((10, 5))
+    input_data = np.random.RandomState(7331).random((1, 10)) - 0.5
+
+    torch_input = torch.tensor(input_data, requires_grad = True)
```

Comment:
My understanding is that gradcheck shouldn't have to compare to torch. Can it just compare to tinygrad in this test? (perhaps a different test to compare to torch if you want one?)